core/base_abstractions/distributions.py

```diff
@@ -33,10 +33,14 @@ def icdf(self, value):
         raise Exception("Inverse CDF is not defined for categorical distributions.")
 
     def sample(self, sample_shape=torch.Size()):
-        return super().sample(sample_shape).unsqueeze(-1)
+        res = super().sample(sample_shape)
+        return res.view(*res.shape[:2], -1)  # return [steps, sampler, flattened]
 
     def log_probs(self, actions: torch.LongTensor) -> torch.FloatTensor:
-        return super().log_prob(actions.squeeze(-1)).unsqueeze(-1)
+        res = super().log_prob(
+            actions.squeeze(-1)
+        )  # if single actor, actions have a singleton action dim
+        return res.view(*res.shape[:2], -1)  # return [steps, sampler, flattened]
```

Comment:
What's the benefit of flattening things like this?

Reply:
In storage we'll end up having a single tensor with the actions of all actors in the model, so we can enforce the distribution samples + log_probs to already provide this format. Doing so, we can also avoid having to pass an `action_space` to the initialization of the distributions...